### PR TITLE
lua: add connection():ssl() API

### LINKED ISSUE #3704: Expose Protocol identification API in Lua scripts
*Description*:
While it is possible to identify the type of protocol (H1/H2 etc) and whether TLS is used on a given request using c++ API in filters, Lua does not expose this.  It would be useful to expose this information in Lua so that Lua filter can make some decisions.

### PR DESCRIPTION
**Pull request by dio**
*Description*:
This patch adds `connection():ssl()` to `StreamHandleWrapper`. This
API returns the indication if the underlying connection using SSL or not.

*Risk Level*: Low

*Testing*: 
Unit and integration tests

*Docs Changes*: 
- Add `connection():ssl()` API.

*Release Notes*:
- lua: added `connection()` wrapper and `ssl()` API.

Related PR: https://github.com/envoyproxy/envoy/pull/3760 (merged)
Fixes #3704

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

File: docs/root/configuration/http_filters/lua_filter.rst [hunk 1]
```diff
@@ -403,3 +414,19 @@ __pairs()
 
 Iterates through every *metadata* entry. *key* is a string that supplies a *metadata*
 key. *value* is *metadata* entry value.
+
+.. _config_http_filters_lua_connection_wrapper:
+
+Connection object API
+---------------------
+
+secure()
```
Comment (mattklein123) [inline docs/root/configuration/http_filters/lua_filter.rst]: Thing for the future where I'm guessing people are going to ask for SSL/TLS connection properties in Lua, should we instead return an SSL/TLS object? This object could have no methods right now but the presence of the object or nil could indicate secure? Just thinking for the future. Thoughts?

Comment (dio) [reply docs/root/configuration/http_filters/lua_filter.rst]: Seems like a good idea. Will update it.

Comment (dio) [reply docs/root/configuration/http_filters/lua_filter.rst]: Updated.

**Comment by ramaraochavali** [comment]
@dio Thank you so much for adding this and the protocol API, useful stuff.

File: docs/root/configuration/http_filters/lua_filter.rst [hunk 2]
```diff
@@ -301,6 +301,17 @@ Returns :repo:`information <include/request_info/request_info.h>` related to the
...
+connection()
+^^^^^^^^^^^^
+
+.. code-block:: lua
+
+  connection = handle:connection()
+
+Returns the current request's underlying :repo:`connection <include/envoy/network/connection.h>`.
+
+Returns a :ref:`metadata object <config_http_filters_lua_connection_wrapper>`.
```
Comment (mattklein123) [inline docs/root/configuration/http_filters/lua_filter.rst]: "metadata object" ? Is this a copy/paste error?

Comment (dio) [reply docs/root/configuration/http_filters/lua_filter.rst]: @mattklein123 oops. Sorry about that. Fixed in 22680ac.

**Comment by dio** [comment]
OK, will take a look.

**Pull request merged**